### PR TITLE
Cyberpion sl connector

### PR DESCRIPTION
**Pull request by cyberpion-yotam**
Request to add:
* Cyberpion Sentinel Data Connector
* Cyberpion Overview Workbook

### CODE REVIEW

**Comment by yotamgod** [comment]
I'll fix WorkbooksMetadata.json once I get an OK on the rest of the files (it will keep changing).

File: DataConnectors/CyberpionSecurityLogs.json
Comment (v-rucdu) [inline DataConnectors/CyberpionSecurityLogs.json:16]: Query returns syntax error as **is_open_b** column doe not exist in table.

File: Workbooks/CyberpionOverviewWorkbook.json
Comment (v-rucdu) [inline Workbooks/CyberpionOverviewWorkbook.json:24]: Query returns syntax error as **is_open_b** column does not exist in table.
Please update all the queries in workbook json

**Review by v-rucdu — CHANGES_REQUESTED** [review]

File: DataConnectors/CyberpionSecurityLogs.json
Comment (cyberpion-yotam) [reply DataConnectors/CyberpionSecurityLogs.json]: Fixed

File: DataConnectors/CyberpionSecurityLogs.json
Comment (cyberpion-yotam) [reply DataConnectors/CyberpionSecurityLogs.json:16]: I'm not sure how you are querying, but "is_open_b" is part of the [sample data](https://github.com/cyberpion-yotam/Azure-Sentinel/blob/cyberpion-sl-connector/Sample%20Data/Custom/CyberpionActionItems_CL.json), and is one of the fields of this connector.

File: Workbooks/CyberpionOverviewWorkbook.json
Comment (cyberpion-yotam) [reply Workbooks/CyberpionOverviewWorkbook.json:24]: Same as with the previous comment - "is_open_b" is part of the sample data, and is one of the fields of this connector.

File: Sample Data/Custom/CyberpionActionItems_CL.json
Comment (v-rucdu) [inline Sample Data/Custom/CyberpionActionItems_CL.json:2]: sample data file should start with an object array, like below
**[
    {
      "TenantId": "93fc153b-9eb9-4c04-9e00-b9bbdcb4ae32",**

Please refer any existing sample data file

File: DataConnectors/CyberpionSecurityLogs.json
Comment (v-rucdu) [reply DataConnectors/CyberpionSecurityLogs.json:16]: This column is getting inserted into the **value** column as JSON string because of which it not found. Have added comment on the sample data file with details, please have a look.

File: Workbooks/CyberpionOverviewWorkbook.json
Comment (v-rucdu) [reply Workbooks/CyberpionOverviewWorkbook.json:24]: This column is getting inserted into the **value** column as JSON string because of which it not found. Have added comment on the sample data file with details, please have a look.

File: Sample Data/Custom/CyberpionActionItems_CL.json
Comment (cyberpion-yotam) [reply Sample Data/Custom/CyberpionActionItems_CL.json]: True, I didn't notice - fixed it

File: DataConnectors/CyberpionSecurityLogs.json
Comment (cyberpion-yotam) [reply DataConnectors/CyberpionSecurityLogs.json:16]: Try it now (updated sample data)

**Review by v-rucdu — APPROVED** [review]

**Comment by cyberpion-yotam** [comment]
Now that I got an approval, do I need to do anything  or will the branch be merged by Microsoft?

**Comment by v-rucdu** [comment]
> Now that I got an approval, do I need to do anything or will the branch be merged by Microsoft?

I see there are some merge conflicts in the PR, please resolve them.

There will be another approval needed from the code owner post that they will take care of merging this PR.

**Comment by cyberpion-yotam** [comment]
Merge conflicts are from a file that is constantly updated (I've fixed it in the past here d4b9ba4). I'll fix it again.

**Comment by cyberpion-yotam** [comment]
Error was cause by file HighUrgencyActionItems.yaml:
Template Id:8e0403b1-07f8-4865-b2e9-74d1e83200a4 is not valid Errors:The name 'CyberpionActionItems_CL' does not refer to any known table...

Should I add the Detection in a separate pull request?

**Comment by cyberpion-yotam** [comment]
@v-rucdu 
Again, I just wanted to understand what If you'd prefer the Detection to be in a different pull request, so the KQL query passes validation? (the error is that CyberpionActionItems_CL can't be found in the detection..)

**Comment by cyberpion-yotam** [comment]
I would really appreciate a response

**Comment by v-jayakal** [comment]
> I would really appreciate a response



> @v-rucdu
> Again, I just wanted to understand what If you'd prefer the Detection to be in a different pull request, so the KQL query passes validation? (the error is that CyberpionActionItems_CL can't be found in the detection..)
A separate pull request for Detection can be created, but not required, Can you tell me where you are getting the error about not finding CyberpionActionItems_CL? Since it is custom table, it will be created as part of the deployment.

**Comment by cyberpion-yotam** [comment]
> > I would really appreciate a response
> 
> > @v-rucdu
> > Again, I just wanted to understand what If you'd prefer the Detection to be in a different pull request, so the KQL query passes validation? (the error is that CyberpionActionItems_CL can't be found in the detection..)
> > A separate pull request for Detection can be created, but not required, Can you tell me where you are getting the error about not finding CyberpionActionItems_CL? Since it is custom table, it will be created as part of the deployment.

If you look at the unsuccessful KQL checks, you can reach [this page](https://dev.azure.com/azure/Azure-Sentinel/_build/results?buildId=17868&view=logs&j=32357ac5-2d6d-5aac-0137-811a316a2707&t=7f4c8a01-3f5d-5b67-9108-27515c68556f&l=53).
I found the error at row 40, stating that the new custom table CyberpionActionItems_CL couldn't be found.
From my understanding of what you said - it should be able to find a table from this deployment (seeing as I added a Workbook as well, I don't understand why this should happen.)
Thanks

**Comment by v-rucdu** [comment]
> > > I would really appreciate a response
> > 
> > 
> > > @v-rucdu
> > > Again, I just wanted to understand what If you'd prefer the Detection to be in a different pull request, so the KQL query passes validation? (the error is that CyberpionActionItems_CL can't be found in the detection..)
> > > A separate pull request for Detection can be created, but not required, Can you tell me where you are getting the error about not finding CyberpionActionItems_CL? Since it is custom table, it will be created as part of the deployment.
> 
> If you look at the unsuccessful KQL checks, you can reach [this page](https://dev.azure.com/azure/Azure-Sentinel/_build/results?buildId=17868&view=logs&j=32357ac5-2d6d-5aac-0137-811a316a2707&t=7f4c8a01-3f5d-5b67-9108-27515c68556f&l=53).
> I found the error at row 40, stating that the new custom table CyberpionActionItems_CL couldn't be found.
> From my understanding of what you said - it should be able to find a table from this deployment (seeing as I added a Workbook as well, I don't understand why this should happen.)
> Thanks

In case of custom log tables, you need to provide the table schema at **Azure-Sentinel\.script\tests\KqlvalidationsTests\CustomTables** 
only than the Kql Validation tests passes. Please refer existing [PR](https://github.com/Azure/Azure-Sentinel/pull/1261) and provide table schema

**Comment by cyberpion-yotam** [comment]
> > > > I would really appreciate a response
> > > 
> > > 
> > > > @v-rucdu
> > > > Again, I just wanted to understand what If you'd prefer the Detection to be in a different pull request, so the KQL query passes validation? (the error is that CyberpionActionItems_CL can't be found in the detection..)
> > > > A separate pull request for Detection can be created, but not required, Can you tell me where you are getting the error about not finding CyberpionActionItems_CL? Since it is custom table, it will be created as part of the deployment.
> > 
> > 
> > If you look at the unsuccessful KQL checks, you can reach [this page](https://dev.azure.com/azure/Azure-Sentinel/_build/results?buildId=17868&view=logs&j=32357ac5-2d6d-5aac-0137-811a316a2707&t=7f4c8a01-3f5d-5b67-9108-27515c68556f&l=53).
> > I found the error at row 40, stating that the new custom table CyberpionActionItems_CL couldn't be found.
> > From my understanding of what you said - it should be able to find a table from this deployment (seeing as I added a Workbook as well, I don't understand why this should happen.)
> > Thanks
> 
> In case of custom log tables, you need to provide the table schema at **Azure-Sentinel.script\tests\KqlvalidationsTests\CustomTables**
> only than the Kql Validation tests passes. Please refer existing [PR](https://github.com/Azure/Azure-Sentinel/pull/1261) and provide table schema

That worked!
Might I recommend adding that last step in the documentation for adding new Detections (I don't recall seeing it).

**Comment by v-rucdu** [comment]
> > > > > I would really appreciate a response
> > > > 
> > > > 
> > > > > @v-rucdu
> > > > > Again, I just wanted to understand what If you'd prefer the Detection to be in a different pull request, so the KQL query passes validation? (the error is that CyberpionActionItems_CL can't be found in the detection..)
> > > > > A separate pull request for Detection can be created, but not required, Can you tell me where you are getting the error about not finding CyberpionActionItems_CL? Since it is custom table, it will be created as part of the deployment.
> > > 
> > > 
> > > If you look at the unsuccessful KQL checks, you can reach [this page](https://dev.azure.com/azure/Azure-Sentinel/_build/results?buildId=17868&view=logs&j=32357ac5-2d6d-5aac-0137-811a316a2707&t=7f4c8a01-3f5d-5b67-9108-27515c68556f&l=53).
> > > I found the error at row 40, stating that the new custom table CyberpionActionItems_CL couldn't be found.
> > > From my understanding of what you said - it should be able to find a table from this deployment (seeing as I added a Workbook as well, I don't understand why this should happen.)
> > > Thanks
> > 
> > 
> > In case of custom log tables, you need to provide the table schema at **Azure-Sentinel.script\tests\KqlvalidationsTests\CustomTables**
> > only than the Kql Validation tests passes. Please refer existing [PR](https://github.com/Azure/Azure-Sentinel/pull/1261) and provide table schema
> 
> That worked!
> Might I recommend adding that last step in the documentation for adding new Detections (I don't recall seeing it).

Sure, we will update the documentation.

**Review by v-rucdu — DISMISSED** [review]

**Review by v-jayakal — DISMISSED** [review]

**Comment by v-jayakal** [comment]
@shainw - Can you please review the detection provided in the PR? Thanks.

**Comment by cyberpion-yotam** [comment]
@shainw 
Any news on this?

**Comment by cyberpion-yotam** [comment]
Sure. Will do now.

**Review by v-rucdu — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**
